src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -1351,6 +1380,28 @@ public ServiceConfig apply(ServiceConfig config) {
    */
   @Override
   public ServiceConfig getConfig() {
+    boolean filterWeb = true;
+    if (getName().equals("i01.htmlFilter")) {
+      log.info("hre");
+    }
+    Map<String, List<MRLListener>> listeners = getOutbox().notifyList;
+    List<Listener> newListeners = new ArrayList<>();
+
+    if (filterWeb) {
+      for (String method : listeners.keySet()) {
+        List<MRLListener> list = listeners.get(method);
+        for (MRLListener listener : list)
+          if (!listener.callbackName.endsWith("@webgui-client")) {
```

Comment:
Also should do some transformations on the callback name to remove the ID if it's local

Reply:
All subscriptions are represented as a list of `MRLListener`right? What about adding a new property to identify whether it should be saved to config, with the default of true so only the webgui client would need to adjust its subscriptions? I think JMonkeyEngine uses a similar system of subscriptions, at the very least I see it in the list of routes, so it might need the same sort of filtering

Reply:
Interesting concept ... it feels a little strange though, of having a external service determine if your data should be saved or filtered .. :thinking:

Reply:
Thinking it through a bit further, I don't think basic filters in the config like `ignore: ["*@webgui-client"]` would work, because then they would fail if the ID changes. We could filter based on the platform or service type, but that would get complicated quickly and still have limitations when we add a new service or platform that requires this same filtering, we would have to go through and add the new one to the filters. 

So right now, the solutions I have are:

1. Basic filtering via config, either per service, instance-wide by Runtime config, or both with service config taking precedence. I don't really like this option as stated above, it will break for different IDs
2. More advanced filtering using service type and platform type heuristics, this gets complicated fast and would still need updating if we need to filter something else later. The filter could still be stored in the config though so it wouldn't be too difficult to update
3. Each MRLListener that represents the subscription would determine whether it should be saved via a new property. This has the advantage of being easy to implement and a new service/platform being added would not require any changes to any other config to filter its subscriptions. It would also not be dependent on the runtime ID or service name. The disadvantage though is that, as you said, it moves the responsibility of filtering from the service that is writing the config to the services making the subscriptions. It also means the filtering is technically hard-coded unless we want to add a whitelist, which then brings us back to point 2 but inverted.

Honestly not really sure what the best solution is.. Do you have any other ideas?

Reply:
Not sure, but I figure using it a bit might make more clean options apparent.

Reply:
I've come up with a solution that there is a Service.filter which is a super with global filtering & each service type is allowed to override it or append their form of filtering ... its the best solution I've been able to think of sofar .. :shrug: